bindings/rust/README.md

```diff
@@ -20,6 +20,20 @@ Generating rust bindings can be accomplished by running the `generate.sh` script
 $ ./bindings/rust/generate.sh
 ```
 
+Alternatively, rust bindings can be generated using pre-built libs2n by first compiling libs2n, and then running the `generate.sh` script:
```

Comment:
I think we should be writing this from the perspective of external consumers, who wouldn't be able to run `generate.sh`. They'd just be pulling the generated `s2n-tls-sys` from crates.io.

So it's probably a little bit more useful to include a workflow oriented around "how do I call `cargo build` and use my own libs2n?", and skip the `.generate.sh` steps.

I think it would also be nice to be a bit more verbose about the process.

E.g.
1. you do this be setting environment variables
2. the environment variables point to ...
3. the final artifact will then ...

Reply:
I rewrote the doc from the perspective of crates consumer with more details. 
I believe this documentation is more valuable for crates users, so I added the same information in the readme for s2n-tls-sys crates (assuming this is the readme that is shown on creates page when published).